WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt

```diff
@@ -593,6 +593,14 @@ class WPMainActivityViewModelTest : BaseUnitTest() {
         verify(onFeatureAnnouncementRequestedObserver, times(1)).onChanged(anyOrNull())
     }
 
+    @Test
+    fun `given jetpack app, when app is launched, then feature announcement is not shown`() = test {
```

Comment:
Warning (⚠️): This test seems to be a false positive, since it is using the default `isJetpackApp: Boolean = false` parameter (see `startViewModelWithDefaultParameters()`). Also, there is a problem with this test because no matter if that parameter is `true` or `false` the result will end up being a success. This is because the given preconditions haven't met, see correct precondition within the below tests:
- `show feature announcement when it's available and no announcement was not shown before`, or
- `show feature announcement when it's available and was not shown before`

Suggestion (💡): As with every such test, which specifies in its `given` part that this is specific to the `Jetpack` app, consider adding a counter part test to complete this logically with a `given wordpress app, when ..., then ...`. This will make it more obvious on what the expected behavior is. Also, I would make pass the `isJetpackApp`, `true` or `false` parameter explicitly to make it obvious what is being actioned upon. For example:
```
    @Test
    fun `given wordpress app, when app is launched, then feature announcement is shown`() = test {
        whenever(appPrefsWrapper.featureAnnouncementShownVersion).thenReturn(-1)
        whenever(appPrefsWrapper.lastFeatureAnnouncementAppVersionCode).thenReturn(840)
        whenever(featureAnnouncementProvider.getLatestFeatureAnnouncement(true)).thenReturn(
                featureAnnouncement
        )

        startViewModelWithDefaultParameters(false)
        resumeViewModelWithDefaultParameters()

        verify(onFeatureAnnouncementRequestedObserver).onChanged(anyOrNull())
    }

    @Test
    fun `given jetpack app, when app is launched, then feature announcement is not shown`() = test {
        whenever(appPrefsWrapper.featureAnnouncementShownVersion).thenReturn(-1)
        whenever(appPrefsWrapper.lastFeatureAnnouncementAppVersionCode).thenReturn(840)
        whenever(featureAnnouncementProvider.getLatestFeatureAnnouncement(true)).thenReturn(
                featureAnnouncement
        )

        startViewModelWithDefaultParameters(true)
        resumeViewModelWithDefaultParameters()

        verify(onFeatureAnnouncementRequestedObserver, never()).onChanged(anyOrNull())
    }
```

Reply:
@ParaskP7 - Yep, good catch on the "true/false", I forgot to pass in the true param. 👍  

For the Jetpack test the preconditions are never used - they are short-circuited by the `isJetpackApp` wrapper in the code, so there should be no need to set them in the test. The test is tricky, if the config value isn't set, it doesn't mean the feature will be shown, just that the code will continue down the path of execution. 

So yes, it does seem like a false/positive, but what is the intent?  To make sure the test passes for jetpack condition or make sure it fails for WordPress (not Jetpack)? :) Basically this test should test the `if (!buildConfigWrapper.isJetpackApp)` statement and that it doesn't execute the contents of the `if` - regardless if it shows the feature or not. Yep, not an easy way to do that - forcing the test to fail for WordPress doesn't mean it fails for Jetpack. We could possibly separate the `check` from the `show` method, but that's not easily testable because the methods are private. Interesting dilemma. I put the pre-conditions back into the test until I can come up with a better way to approach this. IMO, the test is fragile either way. 🤷  

BTW, I didn't add the WordPress counter test because there are other tests that check that the feature is shown with the same exact preconditions. However, I see your point of having a separate test with different expectations in the name, so I added it.

Reply:
Yeap, thanks for the details explanation @zwarm ! 🌟